src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/AmazonSqsQueryStringParser.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\AmazonSqs\Transport;
+
+/**
+ * @author Yukihiro Arisawa <goreboothero@gmail.com>
+ */
+class AmazonSqsQueryStringParser
+{
+    public function parse(string $queryString): array
+    {
+        $query = [];
+
+        parse_str($queryString, $query);
+
+        foreach ($query as $key => $value) {
+            $query[$key] = str_replace(' ', '+', $value);
+        }
```

Comment:
Following my previous comment:
```suggestion
        $query['secret_key'] = str_replace(' ', '+', $query['secret_key']);
```